Vic2ToHoI4/Source/HOI4World/Sounds/SoundEffectsFactory.cpp

```diff
@@ -0,0 +1,27 @@
+#include "SoundEffectsFactory.h"
+#include "../../V2World/Country.h"
+#include "SoundEffectsMapperFactory.h"
+
+
+
+HoI4::SoundEffectsFactory::SoundEffectsFactory()
+{
+	soundEffectsMapper = SoundEffectsMapper::Factory{}.getMapper();
+}
+
+
+std::vector<HoI4::SoundEffect> HoI4::SoundEffectsFactory::createSoundEffects(
+	 const std::map<std::string, std::shared_ptr<Country>>& countries) const
+{
+	std::vector<SoundEffect> soundEffects;
+
+	for (const auto& [tag, country]: countries)
+	{
+		for (const auto& mapping: soundEffectsMapper->getMappings(country->getSourceCountry().getPrimaryCulture()))
```

Comment:
And you are certain every country has a primary culture set?

Reply:
For the moment a missing primary culture is an empty string, either mapping to nothing or mappings for undefined cultures.